client.go

```diff
@@ -59,10 +59,19 @@ func New(verifySSL bool, controllerURL string, token string, username string) (*
 	return &Client{
 		HTTPClient:    createHTTPClient(verifySSL),
 		VerifySSL:     verifySSL,
-		ControllerURL: *u,
+		ControllerURL: chooseScheme(u),
 		Token:         token,
 		Username:      username,
 		ResponseLimit: DefaultResponseLimit,
 		UserAgent:     DefaultUserAgent,
 	}, nil
 }
+
+//chooseScheme sets a scheme of http if a scheme is not set.
+func chooseScheme(u *url.URL) *url.URL {
+	if u.Scheme == "" || (u.Scheme != "http" && u.Scheme != "https") {
```

Comment:
this could just be `if u.Scheme != "http" && u.Scheme != "https"` for now. we don't really want to use any other schemes right now, do we?